### PR TITLE
Misc. date annotator and case annotator improvements

### PR DESCRIPTION
**Pull request by nathanathan**
Here is a list of the improvements:
* The date annotator can parse ordinal dates like "The first month of 2016."
* Relative dates when used in date ranges could be incorrectly parsed.
* Backwards date ranges are omitted because they are usually the result of something being incorrectly parsed.
* An additional date pattern was added to pick up "19 Aug 2017" style dates that frequently appear on ProMED-mail.
* Additional synonyms were added to the case count detection patterns.
* A contextual word sense disambiguation algorithm is used to omit some of the false positive singular case counts.

### CODE REVIEW

File: epitator/date_annotator.py [hunk 1]
```diff
@@ -78,13 +158,14 @@ def date_to_datetime_range(text, relative_base=doc.date, prefer_dates_from='past
                 date_spans.append(ne_span)
         # Regex for formatted dates
         regex = re.compile(
+            # date MonthName yyyy
```
Comment (Broham) [inline epitator/date_annotator.py]: Are we sure these are the only date formats we want to match?  Looking at the `date MonthName yyyy` format it occurs to me that when I say my birthday I usually use the format `MonthName date yyyy` - should we be matching for this as well?

Also - the comments say these match on dates formatted with dashes (`-`) but they will also match date formatted with forward slashes(`/`) but not back slashes (`\`).  I'm not sure if it matters, but these will match across line breaks as well.

In summary, these will match:
`11/11/1111`
```
11-
11-
1111
```

and these won't:
`11\11\1111`

Will this be ok in our situation?

Comment (nathanathan) [inline epitator/date_annotator.py]: The SpaCy NER might be able to recognize your pattern. This regex is basically a work around for the pattern not being picked up by the current SpaCy model. I'm not concerned about picking up every pattern the NER misses since many of those errors will go away when we update the SpaCy model. This one is of particular importance because it appears frequently in ProMED articles.

I think matching dates across line breaks might be useful more often than it causes false positives, although it's a rare situation in both cases.

I'm okay with leaving out backslashes since I haven't seen dates formatted like that before.

Comment (Broham) [inline epitator/date_annotator.py]: Cool - merge on then :+1:
